From d5c2d0651b11ae923284ce4d885e38ddb45368d0 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Thu, 14 Jun 2012 19:53:04 +0900
Subject: [PATCH 29/97] ext/dfbvideosink: disable signal handler owned by
 DirectFB

DirectFB kills its own process when it catches an interrupt signal.
This behavior makes GStreamer terminate without normal termination
processing.

This patch passes "--dfb:no-sighandler" option to DirectFBInit()
to disable the signal handler owned by DirectFB. It allows
GStreamer's termination processing to proceed normally.
---
 ext/directfb/dfbvideosink.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index 75f3edc..04ff54e 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -575,8 +575,10 @@ gst_dfbvideosink_setup (GstDfbVideoSink * dfbvideosink)
         "DirectFB fullscreen");
     if (!dfbvideosink->dfb) {
       DFBGraphicsDeviceDescription hw_caps;
-      char *argv[] = { (char *) "-", (char *) "--dfb:quiet", NULL };
-      int argc = 2;
+      char *argv[] = { (char *) "-", (char *) "--dfb:quiet",
+        (char *) "--dfb:no-sighandler", NULL
+      };
+      int argc = 3;
       char **args;
 
       GST_DEBUG_OBJECT (dfbvideosink, "initializing DirectFB");
-- 
1.7.10.4

